kubernetes/resource_kubernetes_storage_class.go

```diff
@@ -101,6 +109,7 @@ func resourceKubernetesStorageClassRead(d *schema.ResourceData, meta interface{}
 	d.Set("storage_provisioner", storageClass.Provisioner)
 	d.Set("reclaim_policy", storageClass.ReclaimPolicy)
 	d.Set("volume_binding_mode", storageClass.VolumeBindingMode)
+	d.Set("allow_volume_expansion", storageClass.AllowVolumeExpansion)
```

Comment:
Thank you for your patience, I'm very new to go programming. I did a bit of reading but not sure if I understood the concept correctly.
- Is the below snippet what is expected? `*`dereferences the pointer. 
```go
err = d.Set("allow_volume_expansion", *storageClass.AllowVolumeExpansion)
if err != nil {
	return err
}
```
- And when you mentioned nil check is it for an error check or to check if `storageClass.AllowVolumeExpansion` itself is nil?

```go
if &storageClass.AllowVolumeExpansion != nil {
	d.Set("allow_volume_expansion", *storageClass.AllowVolumeExpansion)
}
```

Reply:
I believe it should be:
```suggestion
	if storageClass.AllowVolumeExpansion != nil {
		d.Set("allow_volume_expansion", *storageClass.AllowVolumeExpansion)
	}
```

Reply:
Great that you're learning Go and doing meaningful work with it in the meantime!

So, `AllowVolumeExpansion` is a pointer to bool, so to access the bool value you would (just like you did) add a dereferencing operator (the `*` character) in front of it. The problem arises when this value is `nil`, because dereferencing a nil will cause Go to panic and the provider process to crash.
To avoid that, you have to first check that the value of `AllowVolumeExpansion` pointer is not `nil` and only in that case access the bool value behind the pointer.

An example of a similar case is seen here:https://github.com/terraform-providers/terraform-provider-kubernetes/blob/965dfd0dabc9488f897c5bfa1086618cf64398bc/kubernetes/structures_pod.go#L14-L16
In this case `ActiveDeadlineSeconds` is a pointer to `int64`. Your nil check can look identical to this.